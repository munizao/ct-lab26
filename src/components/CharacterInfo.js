import React from 'react';
import styles from './App.css';
import PropTypes from 'prop-types';
import InfoLine from './InfoLine';


const CharacterInfo = (props) => (
  <div className={styles.CharacterInfo}>
    <InfoLine lineHead="STATUS" value={props.status}/>
    <InfoLine lineHead="SPECIES" value={props.species}/>
    <InfoLine lineHead="GENDER" value={props.gender}/>
    <InfoLine lineHead="ORIGIN" value={props.origin}/>
    <InfoLine lineHead="LAST LOCATION" value={props.lastLocation}/>
  </div>
);

CharacterInfo.propTypes = {
  status: PropTypes.string,
  species: PropTypes.string,
  gender: PropTypes.string,
  origin: PropTypes.string,
  lastLocation: PropTypes.string,
};

export default CharacterInfo;
