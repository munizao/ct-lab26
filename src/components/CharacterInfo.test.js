import React from 'react';
import { shallow } from 'enzyme';
import CharacterInfo from './CharacterInfo';

describe('CharacterInfo component', () => {
  it('renders CharacterInfo', () => {
    const wrapper = shallow(<CharacterInfo status="Alive" species="Human" gender="Female" origin="Earth" lastLocation="Mars" />);
    expect(wrapper).toMatchSnapshot();
  });
});

