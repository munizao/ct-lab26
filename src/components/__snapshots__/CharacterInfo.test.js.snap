// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterInfo component renders CharacterInfo 1`] = `
<Fragment>
  <InfoLine
    lineHead="STATUS"
    value="Alive"
  />
  <InfoLine
    lineHead="SPECIES"
    value="Human"
  />
  <InfoLine
    lineHead="GENDER"
    value="Female"
  />
  <InfoLine
    lineHead="ORIGIN"
    value="Earth"
  />
  <InfoLine
    lineHead="LAST LOCATION"
    value="Mars"
  />
</Fragment>
`;
